src/multi.c

```diff
@@ -100,6 +100,9 @@ void multiCommand(client *c) {
         return;
     }
     c->flags |= CLIENT_MULTI;
+
+    /* we do not want to allow blocking commands inside multi */
+    c->flags |= CLIENT_DENY_BLOCKING;
```

Comment:
maybe assert that it's not already on?
otherwise, i'm still paranoid that discardTransaction will turn it off in case of reentrance.

Reply:
You can not reenter multi when inside multi, but sure I will assert it.